inc/type.class.php

```diff
@@ -200,13 +200,64 @@ function post_addItem() {
    }
 
    function prepareInputForUpdate($input) {
+      // Handle impact icon uploads
+      $icon = $input['_impact_icon'][0] ?? null;
+      if ($icon) {
```

Comment:
For security reasons, you should check that file is located inside `GLPI_TMP_DIR`.

```suggestion
      $icon = isset($input['_impact_icon'][0]) ? realpath($input['_impact_icon'][0])) : false;
      if ($icon !== false && str_starts_with($icon, realpath(GLPI_TMP_DIR))) {
```